src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/FilePathNormalizer.cs

```diff
@@ -29,7 +29,7 @@ public string Normalize(string filePath)
                 return "/";
             }
 
-            var decodedPath = WebUtility.UrlDecode(filePath);
+            var decodedPath = filePath.Contains("%") ? WebUtility.UrlDecode(filePath) : filePath;
```

Comment:
I get the intent but ugh, feels like a hack and I'm worried other encoded/decoded scenarios might break (I really don't know what else the encoding does). @blowdart Is there a better way for us to conditionally decode file paths so we don't lose `+` signs in file paths after multiple decodes?

Reply:
@Tratcher, save us!!! Is there a better way? ❤️

Reply:
There is no conclusive way to sniff the content and figure out if it was already decoded because it could have double escaping (%2520), you need to track the encoding/decoding state independently either as a separate field or as part of your control flow.

Reply:
I see, well bummer! Thanks for the insight!

Reply:
@Tratcher yeah, the only solutions I could come up with were:
1. This, take a best-guess at it. I don't THINK there are any current scenarios where we might have double-encoded strings, so I'm hoping it's good enough.
2. Have everywhere that is currently taking a path as a string instead take a URI wrapper with a property indicating if encoding has already been run on it (implicit conversion to string for API usefulness).
3. Remove ALL calls to `Normalize` and anything else that runs `WebUtility.UrlDecode` except at the Endpoint layer.

Given those options we're going to chose the one I've already written, but it's good to have the discussion on record in case we ever have a reason we need to come back to this.